tv_apps/tv-deck/js/channel_manager.js

```diff
@@ -25,7 +30,305 @@
+        return this._compareChannel(channelA.number, channelB.number);
+      }.bind(this));
+
+      var i;
+      for (i = 0; i < channels.length; i++) {
+        this.playingSource.channelIndexHash[channels[i].number] = i;
+        this.playingSource.channels[i] = {
+          channel: channels[i],
+          programs: {},
+        };
+      }
+
+      if (!this.playingChannelNumber) {
+        this.playingChannelNumber = channels[0].number;
+      }
+
+      if (!this.playingChannel) {
+        this.fire('error');
+        return;
+      }
+
+      this.fire('tune');
+    }.bind(this), function onerror(error) {
+      alert(error);
+    });
+  };
+
+  proto.setPlayingSource = function cm_setPlayingSource(callback) {
+    if (this.isScanning) {
+      return;
+    }
+
+    if (this.playingTuner.tuner.currentSource === this.playingSource.source) {
+      this.setPlayingChannel(callback);
+      return;
+    }
+
+    this.playingTuner.tuner.setCurrentSource(this.playingSourceType)
+      .then(function() {
+        this.setPlayingChannel(callback);
+      }.bind(this), function() {
+        alert('Source not found');
+      });
+  };
+
+  proto.setPlayingChannel = function cm_setPlayingChannel(callback) {
+    if (this.isScanning) {
+      return;
+    }
+
+    this.playingSource.source.setCurrentChannel(this.playingChannelNumber)
+      .then(function() {
+        if (callback) {
+          callback();
+        }
+      }.bind(this), function() {
+        alert('Channel not found');
+      });
+  };
+
+  proto._compareChannel = function cm__compareChannel(channelA, channelB) {
+    var dashIndexA = channelA.indexOf('-');
+    var dashIndexB = channelB.indexOf('-');
+    var numberA = (dashIndexA >= 0) ?
+                        channelA.substring(0, dashIndexA) : channelA;
+    var numberB = (dashIndexB >= 0) ?
+                        channelB.substring(0, dashIndexB) : channelB;
+    numberA = parseInt(numberA, 10);
+    numberB = parseInt(numberB, 10);
+
+    if (numberA === numberB) {
+      if (dashIndexA === -1 && dashIndexB === -1) {
+        return 0;
+      } else if (dashIndexA !== -1 && dashIndexB === -1) {
+        return 1;
+      } else if (dashIndexA === -1 && dashIndexB !== -1) {
+        return -1;
+      }
+
+      channelA = channelA.substring(dashIndexA + 1);
+      channelB = channelB.substring(dashIndexB + 1);
+      return this._compareChannel(channelA, channelB);
+    }
+    return numberA - numberB;
+  };
+
+  proto.switchChannel = function cm_switchChannel(direction) {
+    if (!this.playingChannel || this.isScanning) {
+      return false;
+    }
+
+    var playingChannelNumber = this.playingChannelNumber;
+    var channelList = this.playingSource.channels;
+    var channelIndex = this.playingSource
+                           .channelIndexHash[playingChannelNumber];
+    if (channelIndex!== 0 && !channelIndex) {
+      return false;
+    }
+
+    switch(direction) {
+      case 'up':
```

Comment:
Please declare `'up', 'down'` as constant variables. The API user can use those constant variables to call this API instead of pure string.